cmd/update_configuration/vers43/datastructure.go

```diff
@@ -60,6 +60,6 @@ type NetDevs struct {
 	Prefix  string            `yaml:"prefix,omitempty"`
 	Netmask string            `yaml:"netmask,omitempty"`
 	Gateway string            `yaml:"gateway,omitempty"`
-	Default string            `yaml:"default,omitempty"`
```

Comment:
We should leave out this config string as this will render old configurations broken. So we should read in this value, but store it in as Primary and so not save it.

Reply:
I haven't looked too much into the update_configuration tool. Should I just delete that one line?

Reply:
They need not match exactly.  There are two data definition in warewulf, which is the `nodeYaml`, which comes an goes to disk, and the datastore, which is of type `NodeInfo`. At the start the `nodeYaml` is parsed and all the datastore of is populated with it. If the storage of `nodes.con` is triggered, a new `nodeYaml` is created from the values of the datastore. So we will have to read in `Default` and use its value for the initialization of the Primay element in the datastore constructor https://github.com/hpcng/warewulf/blob/3edb8d5294f7495ebb04a56adbcd1b02d8d7895e/internal/pkg/node/constructors.go#L52
but leave it out in the code which stores the datastore in https://github.com/hpcng/warewulf/blob/3edb8d5294f7495ebb04a56adbcd1b02d8d7895e/internal/pkg/node/modifiers.go#L52

Reply:
Actually, let me restate this. To update the file, we are concerned only with the structure of nodes.conf (nodeYaml). If the file is older, you read it using the correct older datastructure, modify it (looping through each intervening version), and write it using the current datastructure. Then you go back to normal operation and read nodes.conf again.
There should never be another datastructure defined for the current release. That's already defined in the nodes package.
Older datastructures should be full copies that shipped with those releases.

Reply:
The file cmd/update_configuration/vers43/datastructure.go should not exist. We need to load the output datastructure directly from internal/pkg/node.
Anytime we need to keep two structs in sync manually, we probably doing it wrong.

Reply:
> The file cmd/update_configuration/vers43/datastructure.go should not exist. 
We can remove it, as the resulting tool is not in the `all` target of the make, just in the update config, but still it has documentation purpose.